cmd/notation/login.go

```diff
@@ -115,7 +115,13 @@ func runLogin(ctx context.Context, opts *loginOpts) error {
 		// ErrPlaintextPutDisabled returned by Login() indicates that the
 		// credential is validated but is not saved because there is no native
 		// credentials store available
-		if savedCred, err := credsStore.Get(ctx, registryName); err != nil || savedCred != cred {
+		credKeyName := registryName
+		if registryName == "docker.io" {
+			// it's as expected that credentials for "docker.io" are stored
+			// under the key "https://index.docker.io/v1/"
+			credKeyName = "https://index.docker.io/v1/"
```

Comment:
How about saving `https://index.docker.io/v1/` in a `const` at the top of the file?

Reply:
Are there any other usecase where registry hostname might be different from how key name? If so probably we should create map for this mapping.

Reply:
@priteshbandi This is the only special case.

@Wwwsylvia We may promote this key mapping in the future versions of `oras-credentials-go`.

Reply:
@Two-Hearts I think it's clearer and more concise to use the literal here.

Reply:
@shizhMSFT Good point!